engine/test/e2e/e2e_dm_test.go

```diff
@@ -123,6 +127,82 @@ func TestDMJob(t *testing.T) {
 	testMetrics(re)
 }
 
+func TestDMDumpSyncJob(t *testing.T) {
+	mysqlCfg := util.DBConfig{
+		Host:     "127.0.0.1",
+		Port:     3306,
+		User:     "root",
+		Password: "",
+	}
+	tidbCfg := util.DBConfig{
+		Host:     "127.0.0.1",
+		Port:     4000,
+		User:     "root",
+		Password: "",
+	}
+
+	mysql, err := util.CreateDB(mysqlCfg)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, util.CloseDB(mysql))
+	}()
+	tidb, err := util.CreateDB(tidbCfg)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, util.CloseDB(tidb))
+	}()
+
+	// clean up
+	_, err = tidb.Exec("drop database if exists dm_meta")
+	require.NoError(t, err)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		testSimpleDumpSyncModeTask(t, mysql, tidb, "test1")
+	}()
+	go func() {
+		defer wg.Done()
+		testSimpleDumpSyncModeTask(t, mysql, tidb, "test2")
+	}()
+	wg.Wait()
+
+	// executor metrics
+	metricsURLs := []string{
+		"http://127.0.0.1:11241/metrics",
+		"http://127.0.0.1:11242/metrics",
+		"http://127.0.0.1:11243/metrics",
+	}
+
+	testMetrics := func(re *regexp.Regexp) {
+		jobIDs := map[string]struct{}{}
+
+		ctx := context.Background()
+		cli, err := httputil.NewClient(nil)
+		require.NoError(t, err)
+		for _, metricsURL := range metricsURLs {
+			resp, err := cli.Get(ctx, metricsURL)
+			require.NoError(t, err)
+			content, err := io.ReadAll(resp.Body)
+			require.NoError(t, err)
+			matched := re.FindAllSubmatch(content, -1)
+			for _, m := range matched {
+				jobIDs[string(m[1])] = struct{}{}
+			}
+			require.NoError(t, resp.Body.Close())
+		}
+		require.Equal(t, 2, len(jobIDs))
+	}
+
+	// test metrics for syncer
+	re := regexp.MustCompile(`syncer.*\{job_id="(.{36})"`)
```

Comment:
our alert on cloud requires all underlying jobs of a single DM job(from dataflow-service view) has the same job-id(dataflow-service side), if we give each underlying job a different id, even with same prefix, like `123-dump` `123-load&sync`, we need to change it's label, like in here to make current alert works. https://stackoverflow.com/questions/70177067/prometheus-substring-or-split-functionality

Reply:
What about using the same job_id for dump and load&sync? Two tasks shouldn't run at the same time

Reply:
fow now, the job is not delete physically for meter purpose, there's still meta left on tiflow-metadb, so same id cannot be used twice